clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java

```diff
@@ -314,6 +319,23 @@ private <T> T execute(
       Map<String, String> headers,
       Consumer<ErrorResponse> errorHandler,
       Consumer<Map<String, String>> responseHeaders) {
+
+    if (beforeConnectHandle != null) {
+      synchronized (this) {
+        if (beforeConnectHandle != null) {
+          Function<HTTPClient, Void> handle = beforeConnectHandle;
+          beforeConnectHandle = null;
+          try {
+            handle.apply(this);
+          } catch (Exception e) {
+            // process until it succeeds
```

Comment:
What's the meaning of this comment?

Reply:
I have added comments
‘The beforeConnectHandle will continue processing until it succeeds.’